goinp/goinp.go

```diff
@@ -63,41 +65,39 @@ func AskForString(messageToPrint string) (string, error) {
 }
 
 func writeStdinClearable(text string) error {
-	for _, c := range []byte(text) {
-		if _, _, errno := syscall.Syscall(syscall.SYS_IOCTL, os.Stdin.Fd(), syscall.TIOCSTI, uintptr(unsafe.Pointer(&c))); errno != 0 {
-			return fmt.Errorf("failed to write to stdin, err no: %d", errno)
+	// clearable stdin text is only possible if there is an available terminal input buffer
+	if terminal.IsTerminal(int(os.Stdin.Fd())) {
+		for _, c := range []byte(text) {
+			if _, _, errno := syscall.Syscall(syscall.SYS_IOCTL, os.Stdin.Fd(), syscall.TIOCSTI, uintptr(unsafe.Pointer(&c))); errno != 0 {
+				return fmt.Errorf("failed to write to stdin, err no: %d", errno)
+			}
 		}
 	}
 	return nil
 }
 
-// AskOptions ...
-func AskOptions(title string, defaultValue string, optional bool, options ...string) (string, error) {
+func askOptionsWithReader(title string, defaultValue string, optional bool, reader *bufio.Reader, options ...string) (string, error) {
 	const customValueOptionText = "<custom value>"
 
 	if len(options) == 0 {
 		for {
 			if title != "" {
 				fmt.Print("Enter value for \"" + strings.TrimSuffix(title, ":") + "\": ")
 			}
-			var (
-				input string
-				err   error
-			)
 
 			if defaultValue != "" {
 				if err := writeStdinClearable(defaultValue); err != nil {
 					return "", err
 				}
 			}
 
-			input, err = bufio.NewReader(os.Stdin).ReadString('\n')
+			input, err := reader.ReadString('\n')
 			if err != nil {
 				return "", err
 			}
 
 			if !optional && strings.TrimSpace(input) == "" {
-				log.Errorf("value must be specified")
+				fmt.Print(colorstring.Red("value must be specified") + "\n")
```

Comment:
why did you replace the log package? both writes to `os.Stdout`

Reply:
If I use log package then I always need to set thats output writer and restore it after I am not using it. That way using my method would globally affect the behavior of the log package.